diff --git a/OS/src/OSParsers/OSParseosil.tab.cpp b/OS/src/OSParsers/OSParseosil.tab.cpp
index eb22477..96c465e 100644
--- a/OS/src/OSParsers/OSParseosil.tab.cpp
+++ b/OS/src/OSParsers/OSParseosil.tab.cpp
@@ -9070,7 +9070,6 @@ bool parseVariables( const char **p,  OSInstance *osinstance, int* osillineno){
                     vt = returnVarType(attText[0]);
                     if( vt == 0 ) {  osilerror_wrapper( ch,osillineno,"variable type not recognized"); return false;}
                     osinstance->instanceData->variables->var[varcount]->type = attText[0];
-                    if (vt == ENUM_VARTYPE_binary) osinstance->instanceData->variables->var[varcount]->ub = 1.0;
                     delete [] attText;
                     break;
                 case 'l':
@@ -9119,6 +9118,13 @@ bool parseVariables( const char **p,  OSInstance *osinstance, int* osillineno){
                     return false;
                     break;
                 }
+                // reset default upper bound for binary variables
+                if (vt == ENUM_VARTYPE_binary)
+                {
+                    if (osinstance->instanceData->variables->var[varcount]->ub == OSDBL_MAX
+                        && varubattON == false)
+                        osinstance->instanceData->variables->var[varcount]->ub = 1.0;
+                }
                 ch++;
             }
             //
