diff --git a/OS/configure b/OS/configure
index b7fa35d..c6ea837 100755
--- a/OS/configure
+++ b/OS/configure
@@ -37165,7 +37165,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
   CFLAGS=`echo $CFLAGS | sed -e 's/-pedantic-errors//'`
   ac_coin_found_socket_libraries=no
   case "$CC" in
-    cl* | */cl* | CL* | */CL* | icl*)
+    cl | */cl | CL* | */CL* | icl*)
       if test x$ac_cv_header_winsock_h = xyes ; then
        LIBS="wsock32.lib"
        echo "$as_me:$LINENO: checking with \"$LIBS\"" >&5
diff --git a/OS/m4/ac_coin_socket_libs.m4 b/OS/m4/ac_coin_socket_libs.m4
index 5dbbec8..e4e0eeb 100644
--- a/OS/m4/ac_coin_socket_libs.m4
+++ b/OS/m4/ac_coin_socket_libs.m4
@@ -36,7 +36,7 @@ AC_DEFUN([AC_COIN_OS_SOCKET_LIBS],
   CFLAGS=`echo $CFLAGS | sed -e 's/-pedantic-errors//'`
   ac_coin_found_socket_libraries=no
   case "$CC" in
-    cl* | */cl* | CL* | */CL* | icl*)
+    cl | */cl | CL* | */CL* | icl*)
       if test x$ac_cv_header_winsock_h = xyes ; then
        LIBS="wsock32.lib"
        AC_MSG_CHECKING(with "$LIBS")
