diff --git a/OS/data/osilFiles/testMatricesAndCones.osil b/OS/data/osilFiles/testMatricesAndCones.osil
index ac00c2c..a4b8a09 100644
--- a/OS/data/osilFiles/testMatricesAndCones.osil
+++ b/OS/data/osilFiles/testMatricesAndCones.osil
@@ -1110,7 +1110,7 @@
                 </elements>
             </matrix>
         </matrices>
-        <cones numberOfCones="10">
+        <cones numberOfCones="11">
             <nonnegativeCone numberOfColumns="2" numberOfRows="2"></nonnegativeCone>
             <nonpositiveCone numberOfColumns="2" numberOfRows="2"/>
             <quadraticCone numberOfColumns="1" numberOfRows="5"></quadraticCone>
diff --git a/OS/src/OSCommonInterfaces/OSInstance.cpp b/OS/src/OSCommonInterfaces/OSInstance.cpp
index 725c1cc..8cc3bb8 100644
--- a/OS/src/OSCommonInterfaces/OSInstance.cpp
+++ b/OS/src/OSCommonInterfaces/OSInstance.cpp
@@ -7553,7 +7553,7 @@ std::string OrthantCone::getConeInXML()
     {
         ubt = ub[i];
         lbt = lb[i];
-        if (ubt == ub[i+mult] && lbt == lb[i+mult])
+        if (ubt == ub[i+mult] && lbt == lb[i+mult] && i+mult < numberOfRows*numberOfColumns)
         {
             mult++;
         }
