diff --git a/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp b/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
index 757e145..cae54a6 100644
--- a/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
+++ b/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
@@ -626,8 +626,8 @@ void IpoptProblem::finalize_solution(SolverReturn status,
     {
         printf("z_U[%d] = %e\n", i, z_U[i]);
     }
-#endif
     if(osinstance->getObjectiveNumber() > 0) printf("\nObjective value f(x*) = %e\n", obj_value  );
+#endif
 
     int solIdx = 0;
     int numberOfOtherVariableResults;
diff --git a/OS/src/OSSolverService.cpp b/OS/src/OSSolverService.cpp
index 3dec7fa..9a17d83 100644
--- a/OS/src/OSSolverService.cpp
+++ b/OS/src/OSSolverService.cpp
@@ -864,7 +864,7 @@ void solve(OSCommandLine *oscommandline)
                     std::system(ch);
                 }
                 else
-                    osoutput->OSPrint(ENUM_OUTPUT_AREA_main, ENUM_OUTPUT_LEVEL_always, 
+                    osoutput->OSPrint(ENUM_OUTPUT_AREA_main, ENUM_OUTPUT_LEVEL_summary, 
                         "Results written to file " + oscommandline->osrlFile);
             }
             else
diff --git a/OS/src/OSUtils/OSParameters.h b/OS/src/OSUtils/OSParameters.h
index d8af53f..6a469a4 100644
--- a/OS/src/OSUtils/OSParameters.h
+++ b/OS/src/OSUtils/OSParameters.h
@@ -115,7 +115,7 @@ enum ENUM_OUTPUT_LEVEL
      ENUM_OUTPUT_LEVEL_NUMBER_OF_LEVELS    // insert other values above this one...
 };
 
-#define DEFAULT_OUTPUT_LEVEL ENUM_OUTPUT_LEVEL_error
+#define DEFAULT_OUTPUT_LEVEL -1
 
 /**
  *  Enumeration for the different areas that can produce output.
