diff --git a/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp b/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
index f4d35bc..50536d8 100644
--- a/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
+++ b/OS/src/OSSolverInterfaces/OSIpoptSolver.cpp
@@ -626,8 +626,8 @@ void IpoptProblem::finalize_solution(SolverReturn status,
     {
         printf("z_U[%d] = %e\n", i, z_U[i]);
     }
-#endif
     if(osinstance->getObjectiveNumber() > 0) printf("\nObjective value f(x*) = %e\n", obj_value  );
+#endif
 
     int solIdx = 0;
     int numberOfOtherVariableResults;
diff --git a/OS/src/OSSolverService.cpp b/OS/src/OSSolverService.cpp
index 1d62044..1fabd1c 100644
--- a/OS/src/OSSolverService.cpp
+++ b/OS/src/OSSolverService.cpp
@@ -848,7 +848,7 @@ void solve(OSCommandLine *oscommandline)
                     std::system(ch);
                 }
                 else
-                    osoutput->OSPrint(ENUM_OUTPUT_AREA_main, ENUM_OUTPUT_LEVEL_always, 
+                    osoutput->OSPrint(ENUM_OUTPUT_AREA_main, ENUM_OUTPUT_LEVEL_summary, 
                         "Results written to file " + oscommandline->osrlFile);
             }
             else
diff --git a/OS/src/OSUtils/OSOutput.h b/OS/src/OSUtils/OSOutput.h
index e0a38ab..ad63d5f 100644
--- a/OS/src/OSUtils/OSOutput.h
+++ b/OS/src/OSUtils/OSOutput.h
@@ -51,7 +51,7 @@ enum ENUM_OUTPUT_LEVEL
      ENUM_OUTPUT_LEVEL_NUMBER_OF_LEVELS    // insert other values above this one...
 };
 
-#define DEFAULT_OUTPUT_LEVEL ENUM_OUTPUT_LEVEL_error
+#define DEFAULT_OUTPUT_LEVEL -1
 
 /**
  *  Enumeration for the different areas that can produce output.
